example/cripts/example1.cc

```diff
@@ -0,0 +1,242 @@
+  instance.metrics[6] = Metrics::Sum::create("cript.example1.c6");
+  instance.metrics[7] = Metrics::Sum::create("cript.example1.c7");
+  instance.metrics[8] = Metrics::Sum::create("cript.example1.c8"); // This one should resize() the storage
+
+  Bundle::Common::activate().dscp(10).cache_control("max-age=259200");
+}
+
+do_txn_close()
+{
+  borrow conn = Client::Connection::get();
+
+  conn.pacing = Cript::Pacing::Off;
+  CDebug("Cool, TXN close also works");
+}
+
+do_send_request()
+{
+  borrow req = Server::Request::get();
+
+  req["X-Leif"] = "Meh";
+}
+
+do_read_response()
+{
+  borrow resp = Server::Response::get();
+
+  resp["X-DBJ"] = "Vrooom!";
+}
+
+do_send_response()
+{
+  borrow resp = Client::Response::get();
+  borrow conn = Client::Connection::get();
+  string msg  = "Eliminate TSCPP";
+
+  resp["Server"]         = "";        // Deletes the Server header
+  resp["X-AMC"]          = msg;       // New header
+  resp["Cache-Control"]  = "Private"; // Deletes old CC values, and sets a new one
+  resp["X-UUID"]         = UUID::Unique::get();
+  resp["X-tcpinfo"]      = conn.tcpinfo.log();
+  resp["X-Cache-Status"] = resp.cache;
+  resp["X-Integer"]      = 666;
+  resp["X-Data"]         = AsString(txn_data[2]);
+
+  resp["X-ASN"]         = conn.geo.ASN();
+  resp["X-ASN-Name"]    = conn.geo.ASNName();
+  resp["X-Country"]     = conn.geo.Country();
+  resp["X-ISO-Country"] = conn.geo.CountryCode();
+
+  // Setup some connection parameters
+  conn.congestion = "bbr";
+  conn.dscp       = 8;
+  conn.pacing     = 100000;
+
+  // Some file operations (note that the paths aren't required here, can just be strings, but it's a good practice)
+  static const File::Path p1("/tmp/foo");
+  static const File::Path p2("/tmp/secret.txt");
+
+  if (File::Status(p1).type() == File::Type::regular) {
+    resp["X-Foo-Exists"] = "yes";
+  } else {
+    resp["X-Foo-Exists"] = "no";
+  }
+
+  string secret = File::Line::Reader(p2);
+  CDebug("Read secret = {}", secret);
+
+  if (resp.status == 200) {
+    resp.status = 222;
+  }
+
+  borrow url2 = Cache::URL::get();
+
+  CDebug("Cache URL: {}", url2.url());
+  CDebug("Cache Host: {}", url2.host);
+  CDebug("Txn count: {}", conn.count());
+}
+
+do_remap()
+{
+  auto now    = Time::Local::now();
+  borrow req  = Client::Request::get();
+  borrow conn = Client::Connection::get();
+  auto ip     = conn.ip();
+
+  if (CRIPT_ALLOW.contains(ip)) {
+    CDebug("Client IP allowed: {}", ip.string(24, 64));
+  }
+
+  CDebug("Epoch time is {} (or via .epoch(), {}", now, now.epoch());
+  CDebug("Year is {}", now.year());
+  CDebug("Month is {}", now.month());
+  CDebug("Day is {}", now.day());
+  CDebug("Hour is {}", now.hour());
+  CDebug("Day number is {}", now.yearday());
+
+  CDebug("from_url = {}", instance.from_url.c_str());
+  CDebug("to_url = {}", instance.to_url.c_str());
+
+  // Turn off the cache for testing
+  // proxy.config.http.cache.http.set(1);
+  // control.cache.nostore.set(true);
```

Comment:
Setting configuration like this looks pretty convenient. I assume we'll only expose configurable parameters for overridable configurations?

Reply:
That is correct. It builds the .hpp file by examining the overridable files, so it's part of the builds. If you add more overridable variables, they become available here as well.